### DISCUSSION TITLE
Can you make a website?

### DISCUSSION THREAD

**Discussion by onlysharma0852** [BrokenSource/DepthFlow#28] [Ideas]
Recently it is working fine. 

![image (14)](https://github.com/BrokenSource/DepthFlow/assets/127586036/b1cbdc3c-f3ff-40a0-9b08-339ae0256ee7)
same image again

https://github.com/BrokenSource/DepthFlow/assets/127586036/ec7de471-bb43-4ffc-b2d3-c52d48628caa

Lookin fine from the last one but i think i still missing some settings.
i used this command
`depthflow input --image (url | path) main --render -s 2`

so i want to ask you that is it possible to make a opensource website just like "Depthy". That will be so much easier for everyone.

**Comment by Tremeschin** [comment]
The code to resize to the image's resolution was at a bad position and causing some edge cases (again depending on monitor resolution and platform), so I've deliberately unhooked it yesterday after some user on Discord, but it's top priority to fix (I have a much better idea where to place it, just struggling on it)

The video of yours just wasn't automatically resized to 1080x1920, the borders are lack of information, using `(...) main -w 1080 -h 1920 (...)` should work :)

**Comment by Tremeschin** [comment]
About a website, I'm quite sure it would be nearly impossible to run the code locally on the user's browser because of Python (and many other factors), if just acessing an url to use it, quite certainly not in real time, __but__...

I can see a Gradio or Streamlit dashboard working if we run it locally, but it's a lot of bandwidth for only me to code right now, as the architecture needs also better ffmpeg, presets system, caching, reusability, but is the end goal per roadmap

One option is to host myself and make it available, I don't have good spare hardware (a R3 1300X and 1050 ti lol), or cloud host it, but for depthflow the computational costs are significant (hugginface spaces are an alternative, not sure if enough cpu there), and-

-would need mechanisms to detect abuse, subscription status if it's paid, etc. not really my expertise in programming

so, for now I'll keep focusing on the foundations, and if someone's up for making a wrapper on top of current stuff or a beta website I'm up to it!

**Comment by Tremeschin** [comment]
I'm very close to a once and for all fix to this aspect ratio enforcing thing, it's much more stable now

I'm reviewing the code and will test on Windows, in a few hours I'll have it commited (^ if things go right 😓)

**Comment by Tremeschin** [comment]
> The code to resize to the image's resolution was at a bad position and causing some edge cases (again depending on monitor resolution and platform), so I've deliberately unhooked it yesterday after some user on Discord, but it's top priority to fix (I have a much better idea where to place it, just struggling on it)
> 
> The video of yours just wasn't automatically resized to 1080x1920, the borders are lack of information, using `(...) main -w 1080 -h 1920 (...)` should work :)

New code is up wih a much better place to have the resolutions calculations!

Now depthflow enforces an aspect ratio to the scene and any resize respects it, so `depthflow -i image main -r -o ./video` will best-match the input image's aspect ratio to the width and height one sends (1920x1080 by default)

It prioritizes width changes over height, and calculates the other missing component if it's not set, so for example:

- `shaderflow default --ar 16/9 -w 1280`: Calculates `h=720` as it's not set
- `shaderflow default --ar 1`: Sets resolution to `(1920, 1920)` as it prioritizes width and ar=1
- `shaderflow default --ar 16:9 -w 1920 -h 90`: Will use the smallest component of the two, so `(160, 90)`

All of those transfer to depthflow after the `main (args)` or just `depthflow (args)` which implies `main`, the `--ar` setting is set by depthflow's `input` command of the image's ratio

Need to git pull all shaderflow/depthflow/monorepo for the new changes as they interop, maybe a `git pull --recurse-submodules --rebase` could work

Let me know if it works, and any potential edge case I missed, but so far it's much better than previous solutions

**Comment by onlysharma0852** [comment]
I will try and let you know later.
